sway-core/src/type_system/id.rs

```diff
@@ -184,10 +185,15 @@ impl TypeId {
         &self,
         engines: &Engines,
         filter_fn: &F,
+        depth: usize,
     ) -> HashMap<TypeId, Vec<TraitConstraint>>
     where
         F: Fn(&TypeInfo) -> bool,
     {
+        if depth >= 128 {
```

Comment:
I don't really like hardcoding this, maybe we should have a configurable type recursion limit?

Reply:
Let us go with CONST for now.